### PR TITLE
Core design cleanup

### PR DESCRIPTION
**Pull request by Catacola**
## Description of the change

This diff incorporates all of the new components built for the leadership dash over the past few weeks and adds the fit and polish needed for the filter bar.

The design in this diff should match Mocks: https://www.figma.com/file/H08AQJSaAIRxfeMP4iCUPh/UP-MVP?node-id=1521%3A43

Things addressed in this diff:
  - Proper placement of filters on the nav bar
  - Proper width of filters
  - Coloring of dropdown filters
  - Size and placement of text within filters
  - Rename `FiltersBar` to `FilterBar`
  - Convert `FilterBar` and `CoreSelect` to `tsx`
  - Update tests
  - Add a new library to support enzyme in react 17
  - Refactor and flatten the div structure of `FilterBar` to make styling and reasoning more straightforward

Things not addressed:
 - Logo on the top bar
 - Placement of section selector on top bar
 - Media queries for responsive design

Since this is such a hodgepodge of little bits, I'm happy to jump on a call to talk over any parts.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Works towards https://github.com/Recidiviz/pulse-dashboard/issues/777

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/core/__tests__/FilterBar.test.js [hunk 1]
```diff
@@ -16,26 +16,30 @@
...
 import React from "react";
-import { queryAllByAttribute } from "@testing-library/dom";
-import { render } from "@testing-library/react";
-import FiltersBar from "../FiltersBar";
+import { configure, mount } from "enzyme";
+import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
+import FilterBar from "../FilterBar";
 
-const getAllById = queryAllByAttribute.bind(null, "id");
+configure({ adapter: new Adapter() });
```
Comment (daschi) [inline src/core/__tests__/FilterBar.test.js:23]: Should this go into a setupTests file so that it runs before all tests?

**Review by daschi — APPROVED** [review]
Woo - exciting seeing the design come together! I had one question about the enzyme configuration, and some design things I noticed below 👁️ 

**For the things addressed in this diff:**
1. Should we also add `cursor: pointer` to the dropdown options?
`.Core-Select__option:active, .Core-Select__option:hover`

**For things not addressed in this diff but I wanted to keep track of:**

1. I noticed on this class, there's a box shadow on the navigation dropdown that is defined with `rem` and I think we may actually want `px`, because the computed size is actually bigger than the design (1.6px):
`.CoreNavigation ul.nav-left .dropdown:hover`

2. The dropdown arrow looks a little cutoff on my screen:
Screen Shot 2021-03-09 at 4 45 16 PM

3. I think for the CorePageSelector options, we need to add a `color` property for this class so that it doesn't use the standard blue color when it's selected:
`.CorePageSelector--Option-Selected  { color: #25b894; }`

File: src/core/__tests__/FilterBar.test.js (hunk 1 above)
Comment (Catacola) [reply src/core/__tests__/FilterBar.test.js:23]: Brilliant! Done

**Comment by Catacola** [comment]
> **For the things addressed in this diff:**
> 
> 1. Should we also add `cursor: pointer` to the dropdown options?
>    `.Core-Select__option:active, .Core-Select__option:hover`

I didn't even notice that before. Nice catch!

> **For things not addressed in this diff but I wanted to keep track of:**
> 
> 1. I noticed on this class, there's a box shadow on the navigation dropdown that is defined with `rem` and I think we may actually want `px`, because the computed size is actually bigger than the design (1.6px):
>    `.CoreNavigation ul.nav-left .dropdown:hover`

I'll address that in the next diff which will be all about the top bar. More generally, what's our rule around when to use px and when to use rem? I've been going with rem to 1 or 2 digits as much as possible and if something is off by 0.2px, so be it.

> 2. The dropdown arrow looks a little cutoff on my screen:
>    Screen Shot 2021-03-09 at 4 45 16 PM

Noted. Thanks!

> 3. I think for the CorePageSelector options, we need to add a `color` property for this class so that it doesn't use the standard blue color when it's selected:
>    `.CorePageSelector--Option-Selected  { color: #25b894; }`

Good catch! Will do in next diff.

**Comment by daschi** [comment]
> More generally, what's our rule around when to use px and when to use rem? I've been going with rem to 1 or 2 digits as much as possible and if something is off by 0.2px, so be it.

I've been thinking about it in terms of the property it's used for, like in this example, I think we would want box-shadow to be more static, and not respond as much to screen changes. Properties that are used for spacing, font-sizes and element sizes I think should be much more responsive and we should use `rem` for those. Does that makes sense?

**Comment by Catacola** [comment]
I see what you're saying, thanks! I'll keep that in mind as I go forward.

**Comment by daschi** [comment]
@Catacola I forgot to mention in my review that I did have to run this for my server to compile, and just noticed that this is also causing a lint failure on travis:

`yarn add -D @types/react-select`

I'm not sure if that could be lending to the test failures too

**Comment by Catacola** [comment]
Thanks for that tip! I had added it, but I think it got lost in a rebase.

For your fulfillment, it turns out that git was just ignoring one of my files. When I went to git mv it, git said it was untracked. When I went to add it, git said ok, and then said there were no new changes. I wanted to rename the file anyway since it is a core-only component, so I just copied the file to its new location, edited it there, and then deleted the old one. Funnily enough, git recorded it as a move from the old location and now the tests pass. This may be the weirdest git situation I've gotten myself into.

**Pull request merged**